libraries/SITL/SIM_Submarine.cpp

```diff
@@ -235,6 +235,7 @@ void Submarine::update(const struct sitl_input &input)
     calculate_forces(input, rot_accel, accel_body);
 
     update_dynamics(rot_accel);
+    update_external_payload(input);
```

Comment:
what does this do?

Reply:
It adds hooks for ancillary things not directly related to the vehicle dynamics.

We'll probably want to move the simulated i2c device updates out of that at some stage, but that's where they're currently hooked.